src/Tests/check-ports.rb

```diff
@@ -15,7 +15,11 @@
     # print "line: " + line;
     if /^lo server running on (\d+)$/.match(line) then
       sleep 3 # give zyn more time to setup
-      port_checker_rval = system(my_path + "/../../rtosc/port-checker 'osc.udp://localhost:" + $1 + "/'")
+      # Timeout 500 is currently required because MiddleWare can be too busy to reply when it
+      # calls OscilGen::get thousands of times in a row to calculate the whole wave table at once.
+      # This should be changed once MiddleWare will calculate the wave table bit by bit and will
+      # be more responsive.
+      port_checker_rval = system(my_path + "/../../rtosc/port-checker --timeout 500 'osc.udp://localhost:" + $1 + "/'")
```

Comment:
Be prepared for some erroneous comments with this pass through:

IIRC the port checker code was moved around after this PR was mostly finalized, so these paths might need updating? (it might be trivially handled with the merge)

Reply:
Correct, on master, we fixed a lot of these timing issues, this is while this change is not required anymore. I just did a local merge from master and removed the whole check-ports.rb, as it was removed on master.